extensions/wikia/ArticleVideo/bower_components/html5-skin/js/components/controlBar.js

```diff
@@ -0,0 +1,506 @@
+
+    //if mobile and not showing the slider or the icon, extra space can be added to control bar width. If volume bar is shown instead of slider, add some space as well:
+    var volumeItem = null;
+    var extraSpaceVolume = 0;
+
+    for (var j = 0; j < defaultItems.length; j++) {
+      if (defaultItems[j].name == "volume") {
+        volumeItem = defaultItems[j];
+
+        var extraSpaceVolumeSlider = (((volumeItem && this.isMobile && !this.props.controller.state.volumeState.volumeSliderVisible) || volumeItem && Utils.isIos()) ? parseInt(volumeItem.minWidth) : 0);
+        var extraSpaceVolumeBar = this.isMobile ? 0 : parseInt(volumeItem.minWidth)/2;
+        extraSpaceVolume = extraSpaceVolumeSlider + extraSpaceVolumeBar;
+
+        break;
+      }
+    }
+
+
+    //if no hours, add extra space to control bar width:
+    var hours = parseInt(this.props.duration / 3600, 10);
+    var extraSpaceDuration = (hours > 0) ? 0 : 45;
+
+    var controlBarLeftRightPadding = CONSTANTS.UI.DEFAULT_SCRUBBERBAR_LEFT_RIGHT_PADDING * 2;
+
+    for (var k = 0; k < defaultItems.length; k++) {
+
+      // filter out unrecognized button names
+      if (typeof controlItemTemplates[defaultItems[k].name] === "undefined") {
+        continue;
+      }
+
+      //do not show CC button if no CC available
+      if (!this.props.controller.state.closedCaptionOptions.availableLanguages && (defaultItems[k].name === "closedCaption")){
+        continue;
+      }
+
+      //do not show quality button if no bitrates available
+      if (!this.props.controller.state.videoQualityOptions.availableBitrates && (defaultItems[k].name === "quality")){
+        continue;
+      }
+
+      //do not show discovery button if no related videos available
+      if (!this.props.controller.state.discoveryData && (defaultItems[k].name === "discovery")){
+        continue;
+      }
+
+      //do not show logo if no image url available
+      if (!this.props.skinConfig.controlBar.logo.imageResource.url && (defaultItems[k].name === "logo")){
+        continue;
+      }
+
+      if (Utils.isIos() && (defaultItems[k].name === "volume")){
+        continue;
+      }
+
+      // Not sure what to do when there are multi streams
+      if (defaultItems[k].name === "live" &&
+        (typeof this.props.isLiveStream === 'undefined' ||
+        !(this.props.isLiveStream))) {
+        continue;
+      }
+
+      controlBarItems.push(defaultItems[k]);
+    }
+
+    var collapsedResult = Utils.collapse(this.props.componentWidth + this.responsiveUIMultiple * (extraSpaceDuration + extraSpaceVolume - controlBarLeftRightPadding), controlBarItems, this.responsiveUIMultiple);
+    var collapsedControlBarItems = collapsedResult.fit;
+    var collapsedMoreOptionsItems = collapsedResult.overflow;
+    this.moreOptionsItems = collapsedMoreOptionsItems;
+
+    finalControlBarItems = [];
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the "let", "const" or "var" keyword to this declaration of "finalControlBarItems" to make it explicit. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2703)